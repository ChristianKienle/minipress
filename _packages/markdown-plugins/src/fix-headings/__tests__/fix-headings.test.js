// @ts-check
const FixHeadingsPlugin = require('./../')
const HeadingsPlugin = require('./../../headings')
const Markdown = require('@minipress/markdown')

describe('Fix Headings Plugin', () => {
  describe('in combination with the headings plugin', () => {
    it('works', () => {
      const md =
        new Markdown()
        .use(FixHeadingsPlugin)
        .use(HeadingsPlugin)
      const page = { headings: [] }
      const env = {
        page
      }
      expect(md.render(`
# Title

## Chapter 1

<!-- fix-headings-begin -->

# Partial 1

<!-- fix-headings-end -->

## Chapter 2
<!-- fix-headings-begin -->

# Partial 2

<!-- fix-headings-end -->

`, env).html).toMatchSnapshot()

      const { headings } = page
      expect(headings).toHaveLength(5)
      expect(headings[0]).toMatchObject({ text: 'Title', level: 1, slug: 'title' })
      expect(headings[1]).toMatchObject({ text: 'Chapter 1', level: 2, slug: 'chapter-1' })
      expect(headings[2]).toMatchObject({ text: 'Partial 1', level: 3, slug: 'partial-1' })
      expect(headings[3]).toMatchObject({ text: 'Chapter 2', level: 2, slug: 'chapter-2' })
      expect(headings[4]).toMatchObject({ text: 'Partial 2', level: 3, slug: 'partial-2' })
    })
  })

  it('works alone', () => {
    const md = new Markdown()
    md.use(FixHeadingsPlugin)
    const markup = md.render('# hello\n\n<!-- fix-headings-begin -->\n# world\n<!-- fix-headings-end -->\n', { page: {} }).html
    expect(markup).toEqual('<h1>hello</h1>\n<h2>world</h2>\n')
  })
})
