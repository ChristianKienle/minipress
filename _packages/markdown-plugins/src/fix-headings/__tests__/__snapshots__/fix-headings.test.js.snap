// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fix Headings Plugin in combination with the headings plugin works 1`] = `
"<h1 id=\\"title\\"><router-link class=\\"header-anchor\\" to=\\"#title\\" aria-hidden=\\"true\\">#</router-link> Title</h1>
<h2 id=\\"chapter-1\\"><router-link class=\\"header-anchor\\" to=\\"#chapter-1\\" aria-hidden=\\"true\\">#</router-link> Chapter 1</h2>
<h3 id=\\"partial-1\\"><router-link class=\\"header-anchor\\" to=\\"#partial-1\\" aria-hidden=\\"true\\">#</router-link> Partial 1</h3>
<h2 id=\\"chapter-2\\"><router-link class=\\"header-anchor\\" to=\\"#chapter-2\\" aria-hidden=\\"true\\">#</router-link> Chapter 2</h2>
<h3 id=\\"partial-2\\"><router-link class=\\"header-anchor\\" to=\\"#partial-2\\" aria-hidden=\\"true\\">#</router-link> Partial 2</h3>
"
`;
